scripts/generate_key.sh

```diff
@@ -0,0 +1,181 @@
+echo "$PASSPHRASE" > "${KEY_ID}_${KEY_FORMAT}_passphrase.txt"
+if [ $( grep "$PASSPHRASE" "${KEY_ID}_${KEY_FORMAT}_passphrase.txt" ) != "$PASSPHRASE" ];
+then
+  echo "passphrase file was not created properly"
+  exit 1;
+fi
+
+
+# zip
+zip "${KEY_ID}_${KEY_FORMAT}_private.gpg.zip" "${KEY_ID}_${KEY_FORMAT}_private.gpg"
+zip "${KEY_ID}_${KEY_FORMAT}_public.gpg.zip" "${KEY_ID}_${KEY_FORMAT}_public.gpg"
+zip "${KEY_ID}_${KEY_FORMAT}_private.gpg.base64.zip" "${KEY_ID}_${KEY_FORMAT}_private.gpg.base64"
+zip "${KEY_ID}_${KEY_FORMAT}_passphrase.txt.zip" "${KEY_ID}_${KEY_FORMAT}_passphrase.txt"
+
+# copy what needed to home directory
+cp ${KEY_ID}_${KEY_FORMAT}*zip ~/caos_key_generator
+ls -la $WORKDIR
+
+# tests
+# create test dir
+TEST_DIR=$(mktemp -d)
+echo "test directory: ${TEST_DIR}"
+cd $TEST_DIR || exit 1
+
+cp ~/caos_key_generator/${KEY_ID}_${KEY_FORMAT}*zip .
+ls -la .
+
+# test
+TEST_FILE="test_original.txt"
+echo "test for $KEY_FORMAT $EMAIL" > $TEST_FILE
+
+
+#TEST_DIR="/var/folders/3f/n1nkdxrj1yv5jr7jl4fdsgzh0000gn/T/tmp.VqD5gVDX"
+cd $TEST_DIR || exit 1;
+# run docker which installs base64 key in the system and generates signature for a test file
+GNUPGHOME="/root/.gnupg"
+cat > test_sign.sh <<EOF
+#!/usr/bin/env sh
+set -e
+
+apt-get -qq update
+apt-get -qq install -y gpg zip unzip
+
+mkdir -p "${GNUPGHOME}"
+chmod -R 600 "${GNUPGHOME}"
+
+cd /srv/test_dir
+
+unzip "${KEY_ID}_${KEY_FORMAT}_private.gpg.base64.zip"
+unzip "${KEY_ID}_${KEY_FORMAT}_passphrase.txt.zip"
+
+ls -la .
+
+cat ${KEY_ID}_${KEY_FORMAT}_passphrase.txt > "${GNUPGHOME}/gpg-passphrase"
+echo "passphrase-file ${GNUPGHOME}/gpg-passphrase" >> "$GNUPGHOME/gpg.conf"
+echo 'allow-loopback-pinentry' >> "${GNUPGHOME}/gpg-agent.conf"
+echo 'pinentry-mode loopback' >> "${GNUPGHOME}/gpg.conf"
+echo 'use-agent' >> "${GNUPGHOME}/gpg.conf"
+echo RELOADAGENT | gpg-connect-agent
+
+cat ${KEY_ID}_${KEY_FORMAT}_private.gpg.base64 | base64 -d | gpg --batch --import -
+
+echo "===> Signing $TEST_FILE"
+gpg --sign --armor --detach-sig $TEST_FILE
+
+ls -la .
+EOF
+chmod +x test_sign.sh
+docker run --rm -t -v $TEST_DIR:/srv/test_dir ubuntu /srv/test_dir/test_sign.sh
+# run docker which imports public key and verify signature of a file
+
+cat > test_verify.sh <<EOF
+#!/usr/bin/env sh
+set -e
+
+apt-get -qq update
+apt-get -qq install -y gpg zip unzip
+
+cd /srv/test_dir
+
+unzip "${KEY_ID}_${KEY_FORMAT}_public.gpg.zip"
+
+ls -la .
+
+gpg --import "${KEY_ID}_${KEY_FORMAT}_public.gpg"
+
+# trust public key
+#for fpr in \$(gpg --list-keys --with-colons  | awk -F: '/fpr:/ {print \$10}' | sort -u); do  echo -e "5\ny\n" |  gpg --command-fd 0 --expert --edit-key \$fpr trust; done
+./trust_key.sh
+
+echo "===> Verifying $TEST_FILE"
+gpg --verify "$TEST_FILE.asc" "$TEST_FILE"
+
+# todo check exit code
+EOF
+
+echo "for fpr in \$(gpg --list-keys --with-colons  | awk -F: '/fpr:/ {print \$10}' | sort -u); do  echo -e \"5\ny\n\" |  gpg --command-fd 0 --expert --edit-key \$fpr trust; done" > trust_key.sh
```

Comment:
A heredoc with `<<EOF` will help to save you from easy to make mistakes while escaping dollars or quotes.